primitives/polkadot-core/src/lib.rs

```diff
@@ -160,14 +167,94 @@ pub type AccountId = <AccountPublic as IdentifyAccount>::AccountId;
 pub type Nonce = u32;
 
 /// Block type of Polkadot-like chains.
-pub type Block = generic::Block<Header, UncheckedExtrinsic>;
+pub type Block = generic::Block<Header, OpaqueExtrinsic>;
 
 /// Polkadot-like block signed with a Justification.
 pub type SignedBlock = generic::SignedBlock<Block>;
 
 /// The balance of an account on Polkadot-like chain.
 pub type Balance = u128;
 
+/// Unchecked Extrinsic type.
+pub type UncheckedExtrinsic<Call> = generic::UncheckedExtrinsic<AccountId, Call, Signature, SignedExtensions<Call>>;
+
+/// A type of the data encoded as part of the transaction.
+pub type SignedExtra = ((), (), (), sp_runtime::generic::Era, Nonce, (), Balance);
+
+/// Parameters which are part of the payload used to produce transaction signature,
+/// but don't end up in the transaction itself (i.e. inherent part of the runtime).
+pub type AdditionalSigned = (u32, u32, Hash, Hash, (), (), ());
+
+/// A simplified version of signed extensions meant for producing signed transactions
+/// and signed payload in the client code.
+#[derive(PartialEq, Eq, Clone, Debug)]
```

Comment:
`Debug` -> `RuntimeDebug`?

Reply:
It's not supposed to be used in `Runtime` anyway, but yeah, let me switch to `RuntimeDebug`.